test/base/test_ensemble.py

```diff
@@ -51,3 +62,65 @@ def test_ensemble_from_optimization():
     ]
     assert hist_tags == ensemble_hist.vector_tags
     assert ep_tags == ensemble_ep.vector_tags
+
+
+def test_ensemble_prediction_from_hdf5():
+    """
+    Test writing an ensemble prediction to hdf5 and loading it.
+    """
+
+    problem = create_petab_problem()
+
+    optimizer = pypesto.optimize.ScipyOptimizer()
+    result = pypesto.optimize.minimize(
+        problem=problem, optimizer=optimizer,
+        n_starts=2,)
+
+    ensemble_ep = Ensemble.from_optimization_endpoints(
+        result=result, max_size=10
+    )
+
+    # This post_processor will transform the output of the simulation tool
+    # such that the output is compatible with the next steps.
+    def post_processor(amici_outputs, output_type, output_ids):
+        outputs = [
+            amici_output[output_type] if amici_output[AMICI_STATUS] == 0
+            else np.full((len(amici_output[AMICI_T]), len(output_ids)), np.nan)
+            for amici_output in amici_outputs
+        ]
+        return outputs
+
+    amici_objective = result.problem.objective
+    state_ids = amici_objective.amici_model.getStateIds()
+    post_processor_x = partial(
+        post_processor,
+        output_type=AMICI_X,
+        output_ids=state_ids,
+    )
+    # Create pyPESTO predictors for states and observables
+    predictor_x = AmiciPredictor(
+        amici_objective,
+        post_processor=post_processor_x,
+        output_ids=state_ids,
+    )
+    engine = MultiProcessEngine()
+    ensemble_prediction = ensemble_ep.predict(predictor_x,
+                                              prediction_id=AMICI_X,
+                                              engine=engine)
+
+    fn = 'test_file.hdf5'
```

Comment:
maybe better with https://docs.python.org/3/library/tempfile.html
What do you think?

Reply:
do not remember why I have started avoiding it tbh :D